src/carousel.tsx

```diff
@@ -159,28 +163,38 @@ const useStyles = createUseStyles((theme) => ({
-            scrollMargin: ({mobilePageOffsetConfig}) => `${mobilePageOffsetConfig.prev}px`,
-            width: ({itemsPerPageConfig, mobilePageOffsetConfig, gap}) =>
-                // prettier-ignore
-                `calc(${100 / itemsPerPageConfig.mobile}% - ${(mobilePageOffsetConfig.next + mobilePageOffsetConfig.prev + gap) / itemsPerPageConfig.mobile}px)`,
+            width: ({itemsPerPageConfig, gap}) =>
+                `calc(${100 / itemsPerPageConfig.mobile}% + ${gap / itemsPerPageConfig.mobile}px)`,
             '&:first-child': {
-                paddingLeft: ({centered, sideMargin}) => (centered ? 0 : sideMargin),
-                width: ({itemsPerPageConfig, mobilePageOffsetConfig, gap, centered, sideMargin}) =>
+                width: ({itemsPerPageConfig, gap}) =>
                     // prettier-ignore
-                    `calc(${100 / itemsPerPageConfig.mobile}% - ${(mobilePageOffsetConfig.next + mobilePageOffsetConfig.prev + gap) / itemsPerPageConfig.mobile + gap - (centered ? 0 : sideMargin)}px)`,
+                    `calc(${100 / itemsPerPageConfig.mobile}% - ${(gap * (itemsPerPageConfig.mobile - 1)) / itemsPerPageConfig.mobile}px)`,
             },
-            '&:last-child': {
-                paddingRight: ({sideMargin}) => sideMargin,
-                width: ({itemsPerPageConfig, mobilePageOffsetConfig, gap, centered, sideMargin}) =>
+
+            scrollSnapAlign: 'start',
+            scrollMargin: ({mobilePageOffsetConfig}) => `${mobilePageOffsetConfig.prev}px`,
+
+            '$hasScroll:not($centered) &': {
+                width: ({itemsPerPageConfig, mobilePageOffsetConfig, gap}) =>
                     // prettier-ignore
-                    `calc(${100 / itemsPerPageConfig.mobile}% - ${(mobilePageOffsetConfig.next + mobilePageOffsetConfig.prev + gap) / itemsPerPageConfig.mobile - (centered ? 0 : sideMargin)}px)`,
+                    `calc(${100 / itemsPerPageConfig.mobile}% - ${(mobilePageOffsetConfig.next + mobilePageOffsetConfig.prev + gap) / itemsPerPageConfig.mobile}px)`,
+                '&:first-child': {
+                    paddingLeft: ({sideMargin}) => sideMargin,
+                    width: ({itemsPerPageConfig, mobilePageOffsetConfig, gap, sideMargin}) =>
+                        // prettier-ignore
+                        `calc(${100 / itemsPerPageConfig.mobile}% - ${(mobilePageOffsetConfig.next + mobilePageOffsetConfig.prev + gap) / itemsPerPageConfig.mobile + gap - sideMargin}px)`,
+                },
+                '&:last-child': {
+                    paddingRight: ({sideMargin}) => sideMargin,
+                    width: ({itemsPerPageConfig, mobilePageOffsetConfig, gap, sideMargin}) =>
+                        // prettier-ignore
+                        `calc(${100 / itemsPerPageConfig.mobile}% - ${(mobilePageOffsetConfig.next + mobilePageOffsetConfig.prev + gap) / itemsPerPageConfig.mobile - sideMargin}px)`,
+                },
```

Comment:
![](https://c.tenor.com/5vo_w_jDfwgAAAAC/calculation-math.gif)

Reply:
perhaps, instead of the carousel controlling the "1" case,  it would be simpler to have two scenarios:
```c
if (itemsCount === 1) 
  render a standalone slide
else
  render carousel
```

Reply:
Not so simple, it's `pageCount`, not `itemsCount`, and you want all the calcs for item width when `items.length < itemsPerPage`. In other words, the mobile carousel should behave like the desktop carousel when there is only one page.